var/spack/repos/builtin/packages/kokkos/package.py

```diff
@@ -355,6 +355,13 @@ def cmake_args(self):
         if self.spec.satisfies("%oneapi") or self.spec.satisfies("%intel"):
             options.append(self.define("CMAKE_CXX_FLAGS", "-fp-model=precise"))
 
+        # Kokkos 4.2.00+ changed the default to Kokkos_ENABLE_IMPL_CUDA_MALLOC_ASYNC=on
+        # which breaks GPU-aware with Cray-MPICH
+        # See https://github.com/kokkos/kokkos/pull/6402
+        # TODO: disable this once Cray-MPICH is fixed
+        if self.spec.satisfies("@4.2.00:") and self.spec["mpi"].name == "cray-mpich":
```

Comment:
Probably needs something like
```suggestion
        if self.spec.satisfies("@4.2.00:") and 'mpi' in self.spec and self.spec["mpi"].name == "cray-mpich":
```